### PR TITLE
Remove dependency on trust-remote-code

### PR DESCRIPTION
**Pull request by anmarques**
This PR is motivated to remove the dependency on the "trust-remote-code" argument for deepsparse transformers pipelines.

In addition to an additional flag, this argument also meant that custom scripts needed for the PyTorch definition of models needed to be carried over in the deployment folder, even though the PyTorch definition is not needed for deepsparse inference.

The dependency was introduced by using the AutoConfig class to load the model config. This PR replaces this with PretrainedConfig instead. AutoConfig is a class that wraps PretrainedConfig and does additional checks to guarantee that the loaded config matches what the model definition uses. By doing this replacement these checks are skipped and the model definitions is not needed any longer.

### Testing plan
 
 The model config at this moment is only used in the text-classification and token-classification pipelines. To validate this PR I evaluated accuracy on these pipelines and verified that it is the same as before. Here are the commands used for testing:

- `deepsparse.transformers.eval_downstream zoo:nlp/token_classification/obert-base/pytorch/huggingface/conll2003/pruned90_quant-none -d conll2003`

- `deepsparse.transformers.eval_downstream zoo:nlp/multilabel_text_classification/obert-base/pytorch/huggingface/goemotions/pruned90_quant-none -d go_emotions`

- `deepsparse.transformers.eval_downstream zoo:nlp/text_classification/obert-base/pytorch/huggingface/mnli/pruned90_quant-none -d mnli`

- `deepsparse.transformers.eval_downstream zoo:nlp/text_classification/obert-base/pytorch/huggingface/qqp/pruned90_quant-none -d qqp`

### Update

@mgoin Pointed out that some models (e.g., replit) can have custom tokenizers and thus trust_remote_code would still be needed for the tokenizer. Hence, updated the code to only remove the argument from the config loading, but still keep it around for the tokenizer.

### CODE REVIEW

File: src/deepsparse/transformers/pipelines/pipeline.py [hunk 1]
```diff
@@ -132,28 +132,30 @@ def setup_onnx_file_path(self) -> str:
         """
         onnx_path = get_onnx_path(self.model_path)
 
+        print(self.config)
```
Comment (dsikka) [inline src/deepsparse/transformers/pipelines/pipeline.py:135]: no prints

File: src/deepsparse/transformers/pipelines/pipeline.py [hunk 2]
```diff
@@ -162,7 +164,6 @@ def setup_onnx_file_path(self) -> str:
 
             self.tokenizer = AutoTokenizer.from_pretrained(
                 self.tokenizer,
-                trust_remote_code=self._trust_remote_code,
                 model_max_length=self.sequence_length,
```
Comment (dsikka) [inline src/deepsparse/transformers/pipelines/pipeline.py:167]: If we're removing this, we have to update the pipeline args as we're now taking in an argument that isn't being used anywhere

**Review by dsikka — CHANGES_REQUESTED** [review]

File: src/deepsparse/transformers/pipelines/pipeline.py (hunk 1 above)
Comment (anmarques) [reply src/deepsparse/transformers/pipelines/pipeline.py]: debugging prints 😁 . Removed

**Review by dbogunowicz — DISMISSED** [review]
great add @anmarques

File: src/deepsparse/transformers/pipelines/pipeline.py (hunk 2 above)
Comment (mgoin) [reply src/deepsparse/transformers/pipelines/pipeline.py:158]: @anmarques it is probably still necessary to allow for the tokenizer for some cases. i had to include this when using replit-code for example https://huggingface.co/replit/replit-code-v1-3b/blob/main/replit_lm_tokenizer.py

**Review by bfineran — APPROVED** [review]

File: src/deepsparse/transformers/pipelines/pipeline.py (hunk 2 above)
Comment (anmarques) [reply src/deepsparse/transformers/pipelines/pipeline.py]: Updated to code to leave trust_remote_code in place as optional argument for tokenizer

**Review by dbogunowicz — APPROVED** [review]
